Task 05.3 Refactoring : User Model

RynniaRyan opened the pull request:
Clean up on user.py included:

- Remove flask-login imports
- Remove type attribute
- Fix which attributes str shows (Fname, Lname, email)
- Rename to_json() to get_json()
- Edit __str__()

RynniaRyan committed:
Task 05.3 Refactoring : User Model

Clean up on user.py included:

- Remove flask-login imports
- Remove type attribute
- Fix which attributes str shows (Fname, Lname, email)
- Rename to_json() to get_json()
- Edit __str__()

JaleneA commented:
1. Have a typo on line 4 -> class User(db.Mode) [Forgot the l in Model]
2. The _init_ uses "staffID" should be userID in this model
3. Edit _str_ to suit
4. Do we need "polymorphic_on': 'type'"?
5. Could include a repr if you want
6. Optional: Clean up the code a bit, spacing out, labelling the attributes with #Attributes, remove excess whitespaces 👍🏽✨
@RynniaRyan

RynniaRyan committed:
Task 05.3 Refactoring : User Model

Updated attributes to include 'type' for distinguishing between Admin and Staff objects.

Will Adjust admin.py accordingly

RynniaRyan committed:
Task 05.3 Refactoring : User Model

Fixed #1 to #3 points made on review by @JaleneA

Edited use of type attribute

RynniaRyan committed:
Task 05.3 Refactoring : User Model

Edited init() parameters - removed userID

JaleneA reviewed (approved):
Seems good, nice job!👍🏽✨ Approved!
